src/header/common/if_match.rs

```diff
@@ -0,0 +1,82 @@
+use header::{EntityTag, Header, HeaderFormat};
+use header::parsing;
+use std::fmt;
+use std::str::FromStr;
+
+/// An entity tag match specification
+#[derive(PartialEq, Clone, Debug)]
+pub enum EntityTagMatch{
+    EntityTags(Vec<EntityTag>),
+    Star
+}
+
+/// The `If-Match` header
+///
+/// The `If-Match` header can be used by clients to indicate what
+/// response matchs they if.
+#[derive(Clone, PartialEq, Debug)]
+pub struct IfMatch(EntityTagMatch);
+
+deref!(IfMatch => EntityTagMatch);
+
+impl Header for IfMatch {
+    fn header_name() -> &'static str {
+        "If-Match"
+    }
+
+    fn parse_header(raw: &[Vec<u8>]) -> Option<IfMatch> {
+        if raw[0][0]==b'*' {
```

Comment:
There isn't a promise that there is always a value. It may be empty, if it was just whitespace and then a newline, for instance. So, this could panic.